sharding/nodeTypeProvider.go

```diff
@@ -0,0 +1,39 @@
+package sharding
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+type nodeTypeProvider struct {
```

Comment:
why does the implementation is done in the sharding package? It is not related.

Reply:
moved